reconcile/dynatrace_token_provider/ocm.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from datetime import timedelta
+from typing import Any
+
+from pydantic import BaseModel
+
+from reconcile.utils.ocm.base import (
+    OCMClusterServiceLogCreateModel,
+)
+from reconcile.utils.ocm.clusters import (
+    ClusterDetails,
+    discover_clusters_by_labels,
+)
+from reconcile.utils.ocm.labels import Filter
+from reconcile.utils.ocm.service_log import create_service_log
+from reconcile.utils.ocm.sre_capability_labels import sre_capability_label_key
+from reconcile.utils.ocm.syncsets import (
+    create_syncset,
+    get_syncset,
+    patch_syncset,
+)
+from reconcile.utils.ocm_base_client import (
+    OCMBaseClient,
+)
+
+"""
+Thin abstractions of reconcile.ocm module to reduce coupling.
+"""
+
+
+class Cluster(BaseModel):
+    id: str
+    external_id: str
+    organization_id: str
+    dt_tenant: str
+
+    @staticmethod
+    def from_cluster_details(cluster: ClusterDetails) -> Cluster:
+        dt_tenant = cluster.labels.get_label_value(
+            f"{sre_capability_label_key('dtp', None)}.tenant"
+        )
+        return Cluster(
+            id=cluster.ocm_cluster.id,
+            external_id=cluster.ocm_cluster.external_id,
+            organization_id=cluster.organization_id,
+            dt_tenant=dt_tenant,
+        )
+
+
+class OCMClient:
```

Comment:
Should this be in utils/ instead so folks can use this pattern to make testing easier going forward?

Reply:
Good question :thinking: 
For now I would like to keep it here, as it also contains DTP specific things (like the Cluster information dedicated for DTP).